python/graphstorm/gconstruct/construct_graph.py

```diff
@@ -646,7 +646,32 @@ def close(self):
         for proc in self.processes:
             proc.join()
 
-def process_node_data(process_confs, remap_id, num_processes):
+def convert2ext_mem(arr, ext_mem_workspace, name):
+    """ Convert a Numpy array to an external-memory Numpy array.
+
+    Parameters
+    ----------
+    arr : Numpy array
+        The input array.
+    ext_mem_workspace : str
+        The path of the directory where the array will be stored.
+    name : str
+        The name of the external memory array.
+
+    Returns
+    -------
+    Numpy array : the Numpy array stored in external memory.
+    """
+    # We need to create the workspace directory if it doesn't exist.
+    os.makedirs(ext_mem_workspace, exist_ok=True)
+    tensor_path = os.path.join(ext_mem_workspace, name + ".npy")
+    em_arr = np.memmap(tensor_path, arr.dtype, mode="w+", shape=arr.shape)
+    em_arr[:] = arr[:]
+    return em_arr
+
+def process_node_data(process_confs, remap_id, num_processes,
+                      ext_mem_workspace=None,
+                      ext_mem_feat_size=64):
```

Comment:
OK, as long as we agree that we expect users to call `process_node_data` and `process_edge_data` we can treat them as top-level entry points functions, and as such they can be expected  to have lots of arguments.

My previous understanding was that since this is a script the users would only use its main function as an entry point and not import these as library functions.